src/artifacts.js

```diff
@@ -175,8 +217,46 @@ api.declare({
   // Construct details for different storage types
   var isPublic = /^public\//.test(name);
   var details  = {};
+  let uploadId;
   switch (storageType) {
+    case 'blob':
+      // Generate the details that we'll be using.
+      details.size = input.size;
```

Comment:
We can do that as an added validation, but I would like the creator to tell us what it thought the overall size is.  This is about verification and validation, I'd rather not compute things we receive from the generating worker where we can easily pass in the value the worker thought it was